app/controllers/submissions_controller.rb

```diff
@@ -436,7 +436,8 @@ def view
       end
     end
 
-    @problemReleased = @submission.scores.pluck(:released).all?
+    @problemReleased = @submission.scores.pluck(:released).all? &&
+                       !@assessment.before_grading_deadline?
```

Comment:
Good catch! I would suggest making sure this is done in annotations_controller too, since I think the viewFeedback view also has a problemReleased check.

Reply:
Good point, verified that students are currently able to view annotations before grading deadline on the viewFeedback view.